### ISSUE TITLE
How can I solve these problem?

### ISSUE THREAD

**Issue by wu-kang66** [Chlumsky/msdf-atlas-gen#94]
![1](https://github.com/Chlumsky/msdf-atlas-gen/assets/157418566/f5b746a8-858e-4c3d-8857-be2159cd60c3)
![2](https://github.com/Chlumsky/msdf-atlas-gen/assets/157418566/71567c7d-f3ff-4c16-a77d-a1b30013f5db)
![3](https://github.com/Chlumsky/msdf-atlas-gen/assets/157418566/71559a46-682d-4900-82aa-558f027e0bc3)

**Comment by Chlumsky** [comment]
Post the font and character codes.

**Comment by Chlumsky** [comment]
Yes, but Skia geometry preprocessing should take care of it. If this happens in a version with Skia preprocessing enabled, I would still be interested in debugging it.

**Comment by wu-kang66** [comment]
@Chlumsky ,OK, I know. This is the font file.
https://drive.google.com/file/d/12GsVxDhk9ZMhfiml1oMyqgLqZxUnv7BD/view?usp=sharing

**Comment by Chlumsky** [comment]
Thanks for bringing this to my attention. I have made some improvements to geometry preprocessing and problems like these should be dramatically reduced now. If you use this font with Skia preprocessing and overlapping contours disabled (it is not needed when Skia is used), there should no longer be any artifacts.

**Comment by wu-kang66** [comment]
@Chlumsky Thank you very much for so serious to the problem.

**Comment by wu-kang66** [comment]
@Chlumsky，is it the Skia preprocessing ? How can I use it?
https://github.com/micycle1/processing-skia

**Comment by Chlumsky** [comment]
No, you just have to make sure that the CMake option `MSDF_ATLAS_USE_SKIA` is enabled when building msdf-atlas-gen and then it should be resolved automatically.

**Comment by wu-kang66** [comment]
How should I do is? is it an option， Which is it?
![2024-05-19_101511](https://github.com/Chlumsky/msdf-atlas-gen/assets/157418566/484a42dd-aa8e-42fa-a972-9c5dbfb578ab)

**Comment by Chlumsky** [comment]
You are still using the old version. First of all, you need to build the version in `master` and make sure that Skia is linked. Then you just run it normally.

**Comment by wu-kang66** [comment]
@Chlumsky Thank you
